From edad1cdf8ce61d7807d10899396115f4652b65f3 Mon Sep 17 00:00:00 2001
From: Alfred Neumayer <dev.beidl@gmail.com>
Date: Sat, 10 Oct 2020 06:00:36 +0200
Subject: [PATCH] Client: Filter out non-qti codecs

Android uses a ranking system to prefer certain codecs over others,
thereas gsthybrissink doesn't make use of it.

Filter out any codecs that would be considered software decoders.

Change-Id: I77596bc36325e5e3fed0dca5cdf2baad5605c03c
---
 codec2/hidl/client/client.cpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/codec2/hidl/client/client.cpp b/codec2/hidl/client/client.cpp
index ff67681..7be6a49 100644
--- a/codec2/hidl/client/client.cpp
+++ b/codec2/hidl/client/client.cpp
@@ -548,16 +548,26 @@ const std::vector<C2Component::Traits>& Codec2Client::listComponents() const {
     }
     Return<void> transStatus = base()->listComponents(
             [this](const hidl_vec<IComponentStore::ComponentTraits>& t) {
+                int removeCounter = 0;
                 mTraitsList.resize(t.size());
                 mAliasesBuffer.resize(t.size());
                 for (size_t i = 0; i < t.size(); ++i) {
+                    if (((std::string)t[i].name).find("c2.qti.") != 0) {
+                        ++removeCounter;
+                        continue;
+                    }
+
                     c2_status_t status = objcpy(
-                            &mTraitsList[i], &mAliasesBuffer[i], t[i]);
+                            &mTraitsList[i - removeCounter], 
+                            &mAliasesBuffer[i - removeCounter],
+                            t[i]);
                     if (status != C2_OK) {
                         ALOGE("listComponents -- corrupted output.");
                         return;
                     }
                 }
+                mTraitsList.resize(t.size() - removeCounter);
+                mAliasesBuffer.resize(t.size() - removeCounter);
             });
     if (!transStatus.isOk()) {
         ALOGE("listComponents -- failed transaction.");
-- 
2.25.1

